core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -286,15 +286,18 @@ class CassandraJournal(cfg: Config)
         result.map(_ => extractTagWrites(serialized))
     }
 
-    tws.onComplete { result =>
-      self ! WriteFinished(pid, p.future)
-      p.success(Done)
-      // notify TagWriters when write was successful
-      result.foreach(bulkTagWrite => tagWrites.foreach(_ ! bulkTagWrite))
+    bulkTagWrite.onComplete { _ =>
+      self ! WriteFinished(pid, writeInProgressForPersistentId.future)
+      writeInProgressForPersistentId.success(Done)
     }(akka.dispatch.ExecutionContexts.sameThreadExecutionContext)
 
     //Nil == all good
-    tws.map(_ => Nil)(akka.dispatch.ExecutionContexts.sameThreadExecutionContext)
+    bulkTagWrite.map(bulkTagWrite => {
```

Comment:
this is https://github.com/akka/akka-persistence-cassandra/pull/468 which I did as a separate PR as not to block the bug fix on getting this build stable